backend/cmm_helpers.mli

```diff
@@ -808,27 +808,31 @@ val placeholder_dbg : unit -> Debuginfo.t
 val placeholder_fun_dbg : human_name:string -> Debuginfo.t
 
 (** Entry point *)
-val entry_point : string list -> phrase
+val entry_point : Compilation_unit.t list -> phrase
 
 (** Generate the caml_globals table *)
-val global_table : string list -> phrase
+val global_table : Compilation_unit.t list -> phrase
 
 (** Add references to the given symbols *)
 val reference_symbols : string list -> phrase
 
-(** Generate the caml_globals_map structure, as a marshalled string constant *)
+(** Generate the caml_globals_map structure, as a marshalled string constant The
+    runtime representation of the type here must match that of [type global_map]
+    in the natdynlink code. *)
```

Comment:
This horrid constraint should probably necessitate a comment in `compilation_unit.ml`, at the definition of type `Name.t`, otherwise someone could change the representation without realising the consequences.

Reply:
Good point. Christ, that's alarming. Added the comment.

Reply:
Added a comment (along with one for `Prefix.t` and the main `t`, since all will appear in marshaled files).